opencog/atomspace/TypeIndex.cc

```diff
@@ -48,6 +48,31 @@ void TypeIndex::removeAtom(AtomPtr a)
 	remove(t, a->getHandle());
 }
 
+size_t TypeIndex::getNumAtomsOfType(Type type, bool count_subclasses) const
+{
+	iterator it(type, count_subclasses);
+	size_t atom_count = 0;
+	it.s = idx.begin();
+	it.send = idx.end();
+
+	// Loop over all the types looking for type and optional subclasses.
+	Type current_type = 0;
+	while (it.s != it.send)
+	{
+		// If this type is a match...
+		if ((type == current_type) || 
+		    (count_subclasses && (classserver().isA(type, it.type))))
+		{
+			// Add the size of the atom set for this type.
+			atom_count += idx.at(current_type).size();
+		}
+		current_type++;
+		++it.s;
+	}
+
```

Comment:
This loop or something similar, appears in many places, One thing I'd like to do, but wanted to ask first, is build subclass indices in the ClassServer so each type has a vector of subclasses so all this iteration isn't necessary just to find the subclasses.

There aren't that many types so the index won't take up much space, and types are generally going to be fairly static when compared to everything else. Typical use case is probably add some new types on startup and never delete any, I'm guessing.